src/tqec/simulation/io_utils/csv_files.py

```diff
@@ -0,0 +1,55 @@
+"""Read and write simulation data from and to csv files."""
+
+import io
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Literal, TextIO
+
+import sinter
+
+
+def write_sinter_stats_to_csv(
+    filepath: str | Path,
+    stats: Iterable[sinter.TaskStats],
+    if_file_exists: Literal["merge"]
+    | Literal["overwrite"]
+    | Literal["raise"] = "raise",
+) -> None:
+    """Write simulation data to csv file.
+
+    This is a convenience function because `sinter` does not expose this functionality. To read
+    simulation data use `sinter.read_stats_from_csv_files`.
+
+    Args:
+        filepath: The file to be written to.
+        stats: List of simulation results. Each element corresponds to a line in a csv file.
+        if_file_exists: How to react if `filepath` points to an existing file.
+            - raise: Raise an `AssertionError` if the file exists.
+            - overwrite: Effectively delete the file and pretend none exists.
+            - merge: Read the file and try to match entries with the simulation data from `stats`
+                by strong id. In case of a match merge the entries.
+    """
+    filepath = Path(filepath)
+
+    if filepath.exists():
+        if if_file_exists == "raise":
+            raise AssertionError(f"File `{filepath}` already exists.")
```

Comment:
Small nitpick: we mostly use single quotes `'` and double quotes `"` in the repository. I do not think we use back-quotes in user-visible messages/documentation (except for formatting purposes).
```suggestion
            raise AssertionError(f"File '{filepath}' already exists.")
```